core/trino-spi/pom.xml

```diff
@@ -229,6 +229,60 @@
                                 </item>
                                 <!-- Backwards incompatible changes since the previous release -->
                                 <!-- Any exclusions below can be deleted after each release -->
+                                <item>
```

Comment:
I think we should go with deprecation step and avoid backward incompatible change here.

Reply:
How would you envision getQueryId method in the security context class? would it return optional or not?
i think it would require spreading out changes over multiple versions  with plugins updating more than once. I think this is the case where lack of compat is justified.

Reply:
Let it be.